practices/src/main/java/com/kingland/practice/loader/FileLoader.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 Kingland Systems Corporation. All Rights Reserved.
+ */
+package com.kingland.practice.loader;
+
+import com.kingland.practice.buffer.BaseBuffer;
+import com.kingland.practice.utils.BusinessException;
+import com.kingland.practice.utils.Common;
+
+import java.io.*;
+
+/**
+ * @author KSC
+ * @description This extends BaseLoader to get data from files
+ */
+public class FileLoader<T> extends BaseLoader {
+    /**
+     * the designation of sending
+     */
+    private final String designation;
+
+    /**
+     * line number recorder
+     */
+    private int lineNumber = 1;
+
+    /**
+     * total line numbers
+     */
+    private final int totalLines;
+
+    /**
+     * Constructor
+     *
+     * @param buffer      set the designation buffer
+     * @param designation the designation of sending
+     */
+    public FileLoader(BaseBuffer<T> buffer, String designation) throws IOException {
+        super(buffer);
+        this.designation = designation;
+        FileReader in = new FileReader(Common.FILE_PATH + designation);
+        LineNumberReader reader = new LineNumberReader(in);
+        String s = reader.readLine();
+        int lines = 0;
+        while (s != null) {
+            lines++;
+            s = reader.readLine();
+        }
+        reader.close();
+        in.close();
+        totalLines = lines;
```

Comment:
PR1: It's time consumimng to collect the total lines by reading the while file, imaging that the file size a 1GB.